src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -726,19 +764,19 @@ private void CoinstakeWorker(CoinstakeWorkerContext context, ChainedBlock chainT
         {
             context.Logger.LogTrace("({0}:'{1}',{2}:{3},{4}:{5})", nameof(chainTip), chainTip, nameof(minimalAllowedTime), minimalAllowedTime, nameof(searchInterval), searchInterval);
 
-            context.Logger.LogTrace("Going to process {0} UTXOs.", context.Coins.Count);
+            context.Logger.LogTrace("Going to process {0} UTXOs.", context.utxoStakeDescriptions.Count);
 
-            // Sort coins by amount, so that highest amounts are tried first
+            // Sort staking UTXOs by amount, so that highest amounts are tried first
             // because they have greater chance to succeed and thus saving some work.
-            List<StakeTx> orderedCoins = context.Coins.OrderByDescending(o => o.TxOut.Value).ToList();
+            List<UtxoStakeDescription> orderedutxoStakeDescriptions = context.utxoStakeDescriptions.OrderByDescending(o => o.TxOut.Value).ToList();
 
             bool stopWork = false;
-            foreach (StakeTx coin in orderedCoins)
+            foreach (UtxoStakeDescription utxoStakeInfo in orderedutxoStakeDescriptions)
             {
-                context.Logger.LogTrace("Trying UTXO from address '{0}', output amount {1}...", coin.Address.Address, coin.TxOut.Value);
+                context.Logger.LogTrace("Trying UTXO from address '{0}', output amount {1}...", utxoStakeInfo.Address.Address, utxoStakeInfo.TxOut.Value);
 
                 // Script of the first coinstake input.
-                Script scriptPubKeyKernel = coin.TxOut.ScriptPubKey;
+                Script scriptPubKeyKernel = scriptPubKeyKernel = utxoStakeInfo.TxOut.ScriptPubKey;
```

Comment:
you have introduced this again, this has been simplified - there is double assignment 
->
`Script scriptPubKeyKernel = utxoStakeInfo.TxOut.ScriptPubKey;`